server/src/controllers/hostingRequests.ts

```diff
@@ -0,0 +1,355 @@
+import {getConnection, getRepository} from "typeorm";
+import {Class} from "../entity/Class";
+import {HostingRequest} from "../entity/HostingRequest";
+import {User} from '../entity/User'
+import {Request, Response} from "express"
+import { Status } from "../enums/status";
+import {insertObjectIntoTable, alterKeys, isTime, validateValues} from "../support/support"
+import { report } from "node:process";
+
+const logger = require('../utils/logger')
+const hostingRequestRouter = require('express').Router()
+
+hostingRequestRouter.delete('/:id', async (request: Request, response: Response) => {
+  const id = parseInt(request.params.id)
+  const connection = await getConnection();
+  const hrRepository = connection.getRepository(HostingRequest)
+  const hr = await hrRepository.findOne({id});
+
+  if (hr == undefined) {
```

Comment:
This is not something you have to change, but do you know that you can evaluate variables as booleans by using ``!!variable``? It work well for objects (``undefined`` and ``null`` check at once), but may be tricky with primitive types (``!!0 === false``, ``!!1 === true``), strings (``!!'' === false``) or arrays (``!![] === true``)